src/imap_stream.rs

```diff
@@ -139,8 +139,8 @@ impl<R: Read + Write + Unpin> ImapStream<R> {
         });
         match res {
             Ok(response) => Ok(Some(response)),
-            Err(rental::RentalError(err, block)) => {
-                self.buffer.return_block(block);
+            Err((err, heads)) => {
```

Comment:
@Voultapher returning the `owner` seems to be missing from `self_cell`

Reply:
This is a bit out of context. I suspect you are looking into using `self_cell` instead of `ouroboros` and some feature is missing. Please elaborate.

Reply:
Yes, in this case with `self_cell` you would just get `err` back, not `block`/`heads`

Looking at the source it seems like `self_cell` just drops everything inside the cell when an error occurs